src/Compilers/Test/Core/Assert/ConditionalFactAttribute.cs

```diff
@@ -159,6 +159,7 @@ public static class ExecutionConditionUtil
         public static bool IsWindowsDesktop => IsWindows && IsDesktop;
         public static bool IsMonoDesktop => Type.GetType("Mono.Runtime") != null;
         public static bool IsMono => MonoHelpers.IsRunningOnMono();
+        public static bool IsMonoRuntime => Type.GetType("Mono.RuntimeStructs") != null;
```

Comment:
Now that we're making a distinction between Mono Core and Mono Framework I would like it if the names we use here are more descriptive. It seems that we essentially want three properties here: 

- `IsMonoDesktop`: true when we are running under the Mono equivalent of .NET Framework
- `IsMonoCore`: true when we are running under the Mono equivalent of .Net Core
- `IsAnyMono`: basically `IsMonoDesktop || IsMonoCore`
- p.

Reply:
hi @jaredpar , Thanks for taking a look. So you are saying to use "IsMonoCore" instead of "IsMonoRuntime" ?

Reply:
Correct. `MonoRuntime` is a bit ambiguous cause it could be the framework or core version.

Reply:
Just to chime in here a bit.  As I understand it, there are four environments we could be running under:
- We could run under .NET Framework (4.x)
- We could run under .NET Core (actually .NET 6, 8, 9,...) using the CoreCLR runtime (i.e. https://github.com/dotnet/runtime/tree/main/src/coreclr)
- We could run under .NET Core (actually .NET 6, 8, 9,...) using the Mono runtime (i.e. https://github.com/dotnet/runtime/tree/main/src/mono)
- We could run under stand-alone Mono itself (i.e. https://github.com/mono/mono)

The distinction @giritrivedi is trying to make here is between the two latter cases.  I'm not sure this is really a question of .NET Framework vs. .NET Core; if I understand correctly, the stand-alone Mono project sort-of supports both of these as well, but that's not what we care about here.

What the test checks for is "are we running under .NET (Core) using the Mono runtime".   [ I don't have strong opinions on any particular name, I just wanted to be sure we're all clear what exactly this is intended to express.  ]


As an aside, I noticed this somewhat strange situation:  in this location, we have:
```
        public static bool IsMonoDesktop => Type.GetType("Mono.Runtime") != null;
        public static bool IsMono => MonoHelpers.IsRunningOnMono();
```
but that helper function is itself defined as:
```
    public static class MonoHelpers
    {   
        public static bool IsRunningOnMono() => Roslyn.Test.Utilities.ExecutionConditionUtil.IsMonoDesktop;
    }
```
so it seems the `IsMono` definition is redundant (always equivalent to `IsMonoDesktop`)?

Reply:
> I'm not sure this is really a question of .NET Framework vs. .NET Core; if I understand correctly, the stand-alone Mono project sort-of supports both of these as well, but that's not what we care about here.

We generally think of `mono/mono` as the framework / desktop mono. Possible this is just our vernacular though. Happy to consider other suffix here. 

> so it seems the IsMono definition is redundant (always equivalent to IsMonoDesktop)?

That is very possible. It's been a while since we had Mono in our core CI loop. I wouldn't be surprised if we had some issues like this that popped up over time.